.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,133 @@
+name: Build & Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-18.04
+
+    steps:
+    - name: Set env
+      run: |
+          echo "::set-env name=KONG_SOURCE_LOCATION::$GITHUB_WORKSPACE"
+          echo "::set-env name=KONG_BUILD_TOOLS_LOCATION::$HOME/kong-build-tools"
+        
+    - uses: azure/docker-login@v1
+      with:
+        username: ${{ secrets.KONG_USERNAME }}
+        password: ${{ secrets.KONG_PASSWORD }}
+
+    - name: Checkout Kong source code
+      uses: actions/checkout@v2
+
+    - name: Setup Kong Build Tools
+      run: make setup-kong-build-tools
+
+    - name: Build Test Image
+      run: pushd $KONG_BUILD_TOOLS_LOCATION && make kong-test-container && popd
```

Comment:
It seems that we are still baking Kong source codes into the Docker container itself, which won't work well for CI environment because we have so many runs with different commit hashes. We should not cache Kong source codes inside Docker hub, but instead run a `make dev` manually inside the container before starting the tests.

Reply:
right now we run `make dev` the precise number of times it's required no more and no less. Running without this cache has downsides:
- speed. Downloading from docker hub saves time over downloading / buildling from source
- reproducible. What one stage tests won't be equivalent to another